Core/Core/Views/HorizontalMenuView.swift

```diff
@@ -0,0 +1,132 @@
+// Copyright © 2019 Instructure, Inc. All rights reserved.
+
+import UIKit
+
+public protocol HorizontalMenuDelegate: class {
+    var maxItemWidth: CGFloat { get }
+    var measurementFont: UIFont { get }
+    var selectedColor: UIColor? { get }
+    func menuItemCount() -> Int
+    func menuItemTitle(at: IndexPath) -> String
+    func didSelectItem(at: IndexPath)
+}
+
+public class HorizontalMenuView: UIView {
+    @IBOutlet private weak var contentView: UIView!
+    @IBOutlet private weak var collectionView: UICollectionView!
+    private var selectedIndexPath: IndexPath? // = IndexPath(item: 0, section: 0)
+    public weak var delegate: HorizontalMenuDelegate?
+    private var cachedTotalWidthOfItems: CGFloat?
+    private var cachedSpaceBetweenCells: CGFloat?
+
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        commonInit()
+    }
+
+    public required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        commonInit()
+    }
+
+    func commonInit() {
+        Bundle.core.loadNibNamed("HorizontalMenuView", owner: self, options: nil)
+        addSubview(contentView)
+        backgroundColor = UIColor.white.ensureContrast(against: .named(.white))
+        contentView.backgroundColor = UIColor.white.ensureContrast(against: .named(.white))
+        contentView.frame = CGRect.zero
+        contentView.pinToAllSidesOfSuperview()
+        collectionView.registerCell(HorizontalMenuViewCell.self, bundle: Bundle.core)
+        collectionView.dataSource = self
+        collectionView.delegate = self
+    }
+
+    override public func layoutSubviews() {
+        super.layoutSubviews()
+        if selectedIndexPath == nil {
+            selectedIndexPath = IndexPath(item: 0, section: 0)
+            selectMenuItem(at: selectedIndexPath)
+        }
+        reload()
+    }
+
+    public func reload() {
+        cachedTotalWidthOfItems = nil
+        cachedSpaceBetweenCells = nil
+        collectionView.reloadData()
+        selectMenuItem(at: selectedIndexPath)
+    }
+
+    public func selectMenuItem(at: IndexPath?, animated: Bool = true) {
+        guard let at = at else { return }
+        collectionView.selectItem(at: at, animated: animated, scrollPosition: UICollectionView.ScrollPosition.centeredHorizontally)
+    }
+}
+
+extension HorizontalMenuView: UICollectionViewDataSource, UICollectionViewDelegate, UICollectionViewDelegateFlowLayout {
+
+    public func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return delegate?.menuItemCount() ?? 0
+    }
+
+    public func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell: HorizontalMenuViewCell = collectionView.dequeue(for: indexPath)
+        cell.title = delegate?.menuItemTitle(at: indexPath)
+        cell.selectionColor = delegate?.selectedColor
+        cell.font = delegate?.measurementFont
```

Comment:
since the font isn't just used for measurement, but is also the display font for the titles, maybe it would be a little clearer to name it `itemFont` or `titleFont` rather than `measurementFont`.